lib/internal/modules/esm/module_job.js

```diff
@@ -1,27 +1,93 @@
 'use strict';
 
 const {
+  ArrayPrototypeFilter,
+  ArrayPrototypeFind,
   ArrayPrototypeJoin,
+  ArrayPrototypeMap,
+  Number,
   ObjectSetPrototypeOf,
   PromiseAll,
   SafeSet,
   SafePromise,
   StringPrototypeIncludes,
   StringPrototypeMatch,
-  StringPrototypeReplace,
   StringPrototypeSplit,
 } = primordials;
 
 const { ModuleWrap } = internalBinding('module_wrap');
 
 const { decorateErrorStack } = require('internal/util');
+const { fileURLToPath } = require('internal/url');
 const assert = require('internal/assert');
 const resolvedPromise = SafePromise.resolve();
 
 function noop() {}
 
 let hasPausedEntry = false;
 
+function extractExample(file, lineNumber) {
+  const { readFileSync } = require('fs');
+  const { parse } = require('internal/deps/acorn/acorn/dist/acorn');
+  const { findNodeAt } = require('internal/deps/acorn/acorn-walk/dist/walk');
+
+  const code = readFileSync(file, { encoding: 'utf8' });
+  const parsed = parse(code, {
+    sourceType: 'module',
+    locations: true,
+  });
+  let start = 0;
+  let node;
+  do {
+    node = findNodeAt(parsed, start);
```

Comment:
What if `findNodeAt` returns `undefined`?

Reply:
Can you imagine a test case that would reproduce this?

Reply:
I'm not sure, but I assume it may happen because it's in your `while` condition.

Have you tried to use `findNodeAround` instead of `findNodeAt`? It seems to fit better our use case.